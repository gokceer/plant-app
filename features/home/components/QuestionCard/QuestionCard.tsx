import { BlurView } from "expo-blur";
import React from "react";
import { ImageBackground, StyleSheet, Text, View } from "react-native";

const QuestionCard = ({ image, title }: { image: string; title: string }) => (
  <View style={styles.container}>
    <ImageBackground source={{ uri: image }} style={styles.image}>
      <View style={styles.questionSection}>
        <BlurView style={styles.blurView}>
          <Text style={styles.questionText}>{title}</Text>
        </BlurView>
      </View>
    </ImageBackground>
  </View>
);

export default QuestionCard;

const styles = StyleSheet.create({
  container: {
    borderRadius: 12,
    width: 240,
    height: 164,
    marginRight: 10,
  },
  image: {
    width: 240,
    height: 164,
    borderRadius: 12,
    overflow: "hidden",
  },
  questionSection: {
    flex: 1,
    borderBottomRightRadius: 12,
    borderBottomLeftRadius: 12,
    overflow: "hidden",
  },
  blurView: {
    marginTop: 100,
    backgroundColor: "rgba(0, 0, 0, 0.2)",
    flex: 1,
  },
  questionText: {
    color: "#fff",
    paddingTop: 10,
    fontFamily: "RubikRegular",
    fontSize: 15,
    paddingHorizontal: 14,
  },
});
