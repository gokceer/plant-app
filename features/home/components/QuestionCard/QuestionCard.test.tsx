import { render } from "@testing-library/react-native";
import React from "react";
import QuestionCard from "./QuestionCard";

jest.mock("expo-blur", () => ({
  BlurView: jest.fn(({ children }) => children),
}));

describe("QuestionCard", () => {
  const mockProps = {
    image: "https://example.com/question-image.jpg",
    title: "Test Question Title",
  };

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<QuestionCard {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
