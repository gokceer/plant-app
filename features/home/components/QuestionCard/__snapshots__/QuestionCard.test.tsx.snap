// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "borderRadius": 12,
      "height": 164,
      "marginRight": 10,
      "width": 240,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "borderRadius": 12,
        "height": 164,
        "overflow": "hidden",
        "width": 240,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://example.com/question-image.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": 164,
            "width": 240,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "RubikRegular",
            "fontSize": 15,
            "paddingHorizontal": 14,
            "paddingTop": 10,
          }
        }
      >
        Test Question Title
      </Text>
    </View>
  </View>
</View>
`;
